common/djangoapps/student/views/dashboard.py

```diff
@@ -629,6 +630,21 @@ def student_dashboard(request):
 
     enterprise_message = get_dashboard_consent_notification(request, user, course_enrollments)
 
+    recovery_email_message = None
+
+    # Update the tag info with 'enterprise_learner' if the user belongs to an enterprise customer.
+    enterprise_learner_data = get_enterprise_learner_data(user=request.user)
+    is_secondary_email_exist = UserProfile.objects.filter(secondary_email=user.email).exists()
```

Comment:
what is this line doing?

Reply:
It does not seem right as `secondary` will not be same as user's primary email. I think you want to do the following.
```
does_secondary_email_exist = bool(user.profile and user.profile.secondary_email)
```